src/main/java/com/aws/emr/profiler/core/persister/S3Persister.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file.
+ * This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aws.emr.profiler.core.persister;
+
+import com.amazonaws.services.s3.transfer.TransferManager;
+import com.amazonaws.services.s3.transfer.TransferManagerBuilder;
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.Optional;
+
+public class S3Persister implements Persister {
+    private static Optional<TransferManager> transferManager;
```

Comment:
Why is this an optional? Doesn't look like we are doing anything that allows an empty value

Reply:
I use optional because I want a way to shut it down and make sure it doesn't exist. I could make it local to the function but then I have to pass it around. Ideally, transfer manager would implement the `AutoClosable` interface so it can be used in a try-with-resources statement.

The ctor with the arg for the transferManager is for testing. I like making dependency injection discrete with ctor's. I find it easier to understand than a framework.